src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx

```diff
@@ -23,68 +23,70 @@ import {
 import getRouteStringFromRoutes from 'app/utils/getRouteStringFromRoutes';
 import theme from 'app/utils/theme';
 
-class DateRange extends React.Component {
-  static getTimeStringFromDate = date => moment(date).local().format('HH:mm');
-
-  static propTypes = {
-    /**
-     * Start date value for absolute date selector
-     * Accepts a JS Date or a moment object
-     *
-     * React does not support `instanceOf` with null values
-     */
-    start: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-
-    /**
-     * End date value for absolute date selector
-     * Accepts a JS Date or a moment object
-     *
-     * React does not support `instanceOf` with null values
-     */
-    end: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-
-    /**
-     * Should we have a time selector?
-     */
-    showTimePicker: PropTypes.bool,
-
-    /**
-     * The maximum number of days in the past you can pick
-     */
-    maxPickableDays: PropTypes.number,
-
-    /**
-     * Use UTC
-     */
-    utc: PropTypes.bool,
-
-    /**
-     * handle UTC checkbox change
-     */
-    onChangeUtc: PropTypes.func,
-
-    /**
-     * Callback when value changes
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * Just used for metrics
-     */
-    organization: SentryTypes.Organization,
-  };
-
+const getTimeStringFromDate = (date: Date) => moment(date).local().format('HH:mm');
+
+type ChangeData = {start?: Date; end?: Date; hasDateRangeErrors?: boolean};
+
+const defaultProps = {
+  showAbsolute: true,
+  showRelative: false,
+  /**
+   * The maximum number of days in the past you can pick
+   */
+  maxPickableDays: MAX_PICKABLE_DAYS,
+};
+
+type Props = {
+  /**
+   * Just used for metrics
+   */
+  organization: LightWeightOrganization;
+
+  /**
+   * Start date value for absolute date selector
+   */
+  start: Date;
+
+  /**
+   * End date value for absolute date selector
+   */
+  end: Date;
+
+  /**
+   * handle UTC checkbox change
+   */
+  onChangeUtc: () => void;
+
+  /**
+   * Callback when value changes
+   */
+  onChange: (data: ChangeData) => void;
+
+  className?: string;
+  /**
+   * Should we have a time selector?
+   */
+  showTimePicker?: boolean;
+
+  /**
+   * Use UTC
+   */
+  utc?: boolean | null;
+} & typeof defaultProps;
```

Comment:
Do you know if the the docblocks are carried over from `defaultProps` when doing this?

It always bothers me a bit that we don't declare all of the types in `Props` when they get sucked in from defaultProps.

Reply:
I don't think they do :cry:

Reply:
We could not use typeof and go with this first approach here:
https://develop.sentry.dev/frontend/defaultprops/#class-components

That should carry over dockblocks. Still does not work well with HoCs I'm afraid.